File: CI/python-coroutine/5/producer-consumer-model.cpp
#include <iostream>
#include <thread>
using namespace std;

int cnt = 0;

void producer(){
    while (true){
        cnt += 1;
    }
}

void consumer(){
    while(true){
        cnt -= 1;
    }
}

int main(){
    std::thread t1(producer);
    std::thread t2(consumer);
    t1.join();
    t2.join();
    std::cout << cnt << std::endl;
}


// compile
// g++ -Wl,--no-as-needed --std=c++11 -pthread producer-consumer-model.cpp -o main

File: CI/python-coroutine/5/producer-consumer-model.py
# -*- encoding=utf-8 -*-


import threading


cnt = 0

def consumer():
    global cnt
    while True:
        cnt -= 1


def producer():
    global cnt
    while True:
        cnt += 1

    
if __name__ == '__main__':

    t1 = threading.Thread(target=producer)
    t2 = threading.Thread(target=consumer)
    t1.start()
    t2.start()